src/azul/plugins/metadata/hca/transform.py

```diff
@@ -81,6 +85,124 @@
 pass_thru_uuid4: PassThrough[api.UUID4] = PassThrough()
 
 
+class OrganismAge(FieldType[JSON, str]):
+
+    def to_index(self, value: Optional[JSON]) -> str:
+        """
+        >>> a = OrganismAge()
+        >>> a.to_index({'value': '20', 'unit': 'year'})
+        '20 year'
+
+        >>> a.to_index({'value': '20', 'unit': None})
+        '20'
+
+        >>> a.to_index(None)
+        '~null'
+
+        >>> a.to_index({'notval': '20', 'notu': 'year'})
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: A dictionary with entries for `value` and `unit` is required
+
+        >>> a.to_index({'value': '', 'unit': 'year'})
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: The `value` entry must not be empty
+
+        >>> a.to_index({'value': '20', 'unit': ''})
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: The `unit` entry must not be empty
+
+        >>> a.to_index({'value': None, 'unit': 'years'})
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: Use `null` for entities with no organism age
+
+        >>> a.to_index({'value': '20 ', 'unit': 'years'})
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: The `value` entry must not contain space characters
+
+        >>> a.to_index({'value': 20, 'unit': 'years'})
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: The `value` entry must be of string type
+
+        >>> a.to_index({'value': '20', 'unit': 'years '})
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: The `unit` entry must not contain space characters
+
+        >>> a.to_index({'value': '20', 'unit': True})
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: The `unit` entry must be of string type
+
+        """
+        if value is None:
+            return NullableString.null_string
+        else:
+            try:
+                value, unit = value['value'], value['unit']
+            except KeyError:
+                raise RequirementError(
+                    'A dictionary with entries for `value` and `unit` is required'
+                )
+            else:
+                reject(value == '', 'The `value` entry must not be empty')
+                reject(unit == '', 'The `unit` entry must not be empty')
+        if value is None:
+            raise RequirementError(
+                'Use `null` for entities with no organism age'
```

Comment:
This message still seems to be taylored to when this method is used to validate filters. I think it simply communicate what's wrong.